cpu_comparisons/umap-cuml.ipynb

```diff
@@ -0,0 +1,497 @@
+VwOe5XXoc+xUXjSf0deeiT/Yy/5n36DywgU0/mQ58sEA4Y4I4390EvvuXYPOoSfljR+pZBKkFKgWI3Ikzl97Mve9ui09tKqiYszJINo1CICxKJukGkPpjaS3V1SUxNCkbRKkukOgQvj1FsJvtqSXmyV0IzOJrukm9kEv+OJg1hF4uuFwx2l9dRaWk0qJbu7FZBIwLSyly9NAllqBKGgZyhqN5rNpAcN/kKDTkTFjGhmv/+nwMpNzBD33/gbBYqHguqX0/e5xkodaEW1Wcr/5DULbd+F+/iXmnnM2OUV5SGI62TW4ZTuqIGCfMZWS236AEo3S//gz+N9cDaqKGgiiOLIIbNxExhxtRCSNRvPVEpfDxOUI6y78GbqlJZgXllC8ZCoR114Ga72MWbqQpBKnY/VW2u58j9SAH3fjIMhQcVYpxrl51P38RfIfn43/o0H6179G7nWTsCWH4f+wBcOMPEoun4kzdDKdJ+xi5MWnYrRbicthvB4vpzx4D29Ouw5duY1UXxQ1qVD363UUnzWDntXb05WUAIsOgilUQUe6x7I01HdBBkGElAIGSPiPdI6O93gwTs3DdtMEjGtSdK/ejlSVCV0hlECKVGvoyIlIqCBA/u/noq/KpGfpByiudL82uT2MY1I13t0HAEh2hghv6Ca134svkkKIyey+4nHiKxMcN2bR5/PBaTSaLyQtYDjGLGNHUfXcssOvyx/4BZ4XX0FfWkzWwgWoySSJljYKLQbK7v8F3oXzyT73LCI7a1EiEUSLmQOXXIVl0vj06EuSBHK6qVuI9hJd/5wWMGg0mi8lRZZZd/7djFx6GiWnTCOQHEBER2e4HknQ40/0MeGppcTUMP6BPqrPL+DQi0/R1Wmn6anVjLrmDALhAYRqI/ryHHJunULwqSbM80pI9IskHtqL92e7yLxuDCadhCjoaPBuxNiaILXTx/pJt2Ety+es2scRBZFgSw/eWA9tA9vpe2UT0bYBRJsBYjKlcxzE/Anc7c2oiaHR82QgmP5ZjscRFBCNOpTw0Pw7ogAyiA4z+nw7ah+ogSRqQiaxz4N8XIjBXZ0A6CrspHxxCBwZmU9XYKHguQUM3rkd/YhMAk82ojSlgwXbN0aACt4/H6TgxBoGtjYi2HXYThuG9YFZ9F61HqnUimDWEfcEPq+PVKPRfEFpAcPnTLJayLvyG4dfZ5975uGfHeeeiXnsKADyr7oM7+tvIWVlIQgCwdVrwGQ8HCwAJHxJxKAFVU5C3w4omIIgGT6/g9FoNJr/MFVR8Na1kF1ThSAI6LNs6KxmQkkP+z3rKLfXkFSiCKkI1e21dOd6GFNxGYIg8Ma0K7HdMY7MiQ6MHVFMvixat+8m+lEPxhOLUVWV3F/O4OCcFzHW5JFxgZNCsYrEkwFCbb2UFc9j26+eZNqtS2nyvE7ZN+YwuLGJ9//8AMkP3cg7/ZjyHVReNJ/try4jFY4iWgBRpHenH53VgKHIiGlUGdFeD0nfx4ZUTSiogGVyPtHmAXLvn4Hv0XoSO/vTrca5ErqMTFKNPlRVRY3L+H6/7/Dm0dUdAIhZVpShcmVPjPDqduSWML6LNxPuG0AqsqL2hwm/chCxwAqKghxPoh+bCQkBuT+G55d7kNsCyAcDIEB8bh/M/hw/ZI1G84WjJS3+D8lcuABDaQkAKY8P35urUKIRlL/O9RCLH7V+1CvTsSuI6j+A2rIaIn2fd5U1Go3mP6r7vZ28e8qP8Dd1IIgic568Ccmop/n+lVh1DnKMZdTkfA29IRO9bRgj8tMTYgIsWP5/zJh7OUVRgcE/7GLT1Q9iOa+coqdOovSWOVhHFVAZGUf5wlkIcQUlnsJbMMihl95noPUAzcZdpMwpNtxyP2O2LkX9Tj6hg90kXumhqHos0x64jlmP/4Cud7YjR2LoyrJwTK3EUpJDKqIQG4gRqusmmYhiOqUUANuF1WRcOQohU480IpPw3h5UWcX3eD2Gmiwcd0xB8cSIb+0nsaWP7LumYl5YRuYN49NzMfwNxRdOj4CUY0TMMeH71R4kWSLa50U/MgvH7ZNRUqBGFNRAEtFhwu/rwRyJMKbcT+jZRpTWAObyI/3lPM/u12Z+1mg0f5cWMPyP0hfkMfyPv8c6aQKFN1yTTkWyW49ax2QRmfTgdSj7XiCVdwqC/dNnfdZoNJr/RS7vFta88xArTr3h8LKi+ZM48cWfoAzTsaN/BQk5hntnE30f1KKi0vrGBnbf8yz+1CCh8tkYrIWHt7UW2zFbzGSHZqIvGEbpslPJK6pm2MIZHD/nYjKsedTe9jg5OS2MP2MG4WVNDD69h/JL52EozyIWDaKrycR2bimxSAtZYg4jd11B0e/mk/2tCeRNHcX7i37CwNYGVFlBdceI7OohNuBD5zCDQUp3YJ5mx3puBeaSHFIdYUKvt2CclEfRswuwzC9Bb7WQ2NKP/dxqEo1eSIGgT1+OYzv6ia5oQ+4IYxjlSM/hoOfoq7UCRTfMIOuEarDoiPV4kYrNKN4E0W3pB0f6vEx0so6sxU4Es46im8eSmwd6u5F4a4hoTwQhy0DWLROJG6IEvQOfwyeu0Wi+qLSUpP9hgpR+umSfPhXnO68eXh5Y/xFJt5uMk+ahy7DRc+dLBHf8hhGvTUEwGFD9LZBRjih+8umURqPRfN7kRJKUEKfLuw1TUzaWolzkEoGBvkOITjvhhJeY248pNxNRr6Nwbg0xOYzDWIxO1DP6urPIvWIiBwM7EduSxAYCFJsmYtNnH7UfteE5kEysPfttbHePQx4rUWSpIvRRO1ufeYjcX89g/g+cGAnC2NMxlI5g642PIvoVRj92Ea2/ehfPY7WMXnE20fWdxGfosBYWojZFaXpsOa2CGX9jB0oqhaKoiJJAtNfL6DvPJ3RKCvePt2Bs85IbDHDgvHeQbCZMUTPKYJzYum5im3tBVUmRBKsO951bUT0JdCMySDUHQAeht9uRiizYLh6BrsBC8E/N+H9Xlw4cjALE05M2dP9
```

Comment:
```suggestion
    "And side-by-side we can see that the separation is very similar between both GPU and CPU versions of UMAP.  Both cluster Ankle Boot, Sneaker, and Sandal together (Green/Yellow/Purple), both separate Trousers and Bags entirely, and both have T-shirts, Shirts, Coats, etc. all mixed together."
```

Reply:
Sorry, I just noticed GitHub does a terrible job in visualization here. The suggestion is a typo fix on line 356, not replacing the image.